online/src/app/products.ts

```diff
@@ -14,3 +14,11 @@ export interface ProductViewModel {
     price: number;
 }
 
+export interface ProductWithoutId {
```

Comment:
A nice trick is to use partial classes. For example in this case, you have Partial<Product>, which lets you build an object without all the properties. OR, the easier version is to put the fields that you won't use as optional (e.g., id?: string - where the question mark sets it as optional and it won't require you to have it).

Reply:
Good to know, I actually ended up not using it but forgot to delete it from the ts file.